rust/lesson-plan.md

```diff
@@ -0,0 +1,315 @@
+
+Here are the resources you should consider turning to first:
+
+- The #rust-training channel on the [TiKV Slack]. This channel exists just for
+  this course. Please consider joining to support your fellow students and other
+  course-takers. There will always be someone there to answer questions, but
+  there may be a delay due to time zones and other factors. Both English and
+  Chinese languages are welcome here.
+
+- The #beginners channel on the official [Rust Discord]. You are almost
+  guaranteed to get some answer here, and if not, don't hesitate to ask again.
+  The people who hang out here are there specifically to help. Because of time
+  zone differences it may take time for somebody to respond. Only English will
+  be consistently understood here.
+
+- The QQ Rust group ([QR code][qq]). For Chinese students, this is one of the
+  major Rust communities in China. This is a great place to hang out generally,
+  and for those unconfident in their English skills, a great place to ask
+  for help.
+
+- The WeChat Rust group #2 ([QR code][wechat]). Again this is mostly for the
+  Chinese audience. With WeChat's group limits there are multiple Rust groups.
+  This is the most recent. It is though relatively unpopulated, so you are more
+  likely to get help in the QQ group. Do join this one though to get in before
+  it spills over into a third group.
+
+These resources may also be helpful:
+
+- The official [users forum]. Apply the "help" tag to your post. Questions
+  usually recieve an answer here, but the responses can be limited.
+
+- [StackOverflow]. Apply the "rust" tag. You may or may not recieve a satisfying
+  answer.
+
+You are also welcome to email the primary author of this course, [Brian
+Anderson][brson], at brian@pingcap.com. I will happily answer questions, and am
+eager to hear your experience with the course.
+
+Finally, if there is a [Rust user group] near you, go check it out. As a Rust programmer
+these groups are where you will build some of your strongest connections. (Note
+that that link goes to the old Rust website and may not be up to date).
+
+
+## Making PNA Rust better
+
+At the end of each project is a link to a survey about your experience. Please
+take a few minutes to complete it, and be comfortable expressing the challenges
+you faced, and your criticisms of the course. The survey results are not seen by
+anybody but the [primary author of the course][author], though aggregate
+statistics may be reported publicly.
+
+As you work through the course content, please be on the lookout for things you
+would improve about the course, and either [submit issues][si] explaining, or
+[submit pull requests][spr] with improvements. (If you are being graded,
+accepted pull requests count toward extra credit during final evaluation &mdash;
+let your instructor or evaluator know! Accepted pull requests to any other
+project used during this course count as well. Note: I have no way to guarantee
```

Comment:
```suggestion
project used during this course should count as well. Note: I have no way to guarantee
```

Reply:
I've reworded this entire awkward phrase to be shorter and less absolute.